undate/undate.py

```diff
@@ -1,22 +1,24 @@
 import datetime
 from calendar import monthrange
 
+from undate.dateformat.base import BaseDateFormat
+
 
 class Undate:
     """Simple object for representing uncertain, fuzzy or partially unknown dates"""
 
+    DEFAULT_FORMAT = "ISO8601"
+
     earliest = None
     latest = None
     label = None
+    formatter = None
 
-    #: datetime strftime format for known part of date
-    iso_format = {
-        "year": "%Y",
-        "month": "%m",
-        "day": "%d",
-    }
+    def __init__(self, year=None, month=None, day=None, formatter=None):
```

Comment:
With an eye to #11 - is `formatter` supposed to be a string or an subclass instance of `BaseDateFormat`?

Reply:
It's meant to be a date format subclass. Should we start adding type annotations or better to wait and do it consistently? I guess at least we could add docstrings

Reply:
I'll start on #11, will open up a new branch based on this PR